### DISCUSSION TITLE
Routing With Dynamic Subdomain For Each user

### DISCUSSION THREAD

**Discussion by ajaytech** [vercel/next.js#14181] [Help]
Hi,

How can the Dynamic Sub domain routing feature be implementing in NextJS?
Example: If a user comes with username **abc** in site **xyz** then he can access his
site on **abc.xyz.com**

Also, if the user have **abc.com**  domain then he can point abc.com to abc.xyz.com
So in future if someone opens **abc.com** then **abc.xyz.com** is served. And in URL
also the abc.com is shown.

I have investigated few plugin in NPM like  **vhost** and **wildcard-subdomains** but not sure that is right way to take on this issue. 

The vhost requires changes in system hosts in local system and wildcard-subdomain solves the issue purely with routing.

Update: 
I have customized **Server.js** With Code:

```
...
if (pathname === "/demo.demo.com") {
      app.render(req, res, "/demo.demo.com", query);
}
...
```
And in **_app.js**
```
static async getInitialProps(appArgument) {
   ...
    return {
      ...
      renderFrom: "demo.demo.com"
    };
  }
```

Also in my host I have demo.demo.com point to localhost.

The site works for me in demo.demo.com:3000 but how to generalise it in production senarios 
with Database and CNAME Records.

**Comment by harishannam** [comment]
@ajaytech DId you get a way to rightly solve this custom subdomain usecase?

**Comment by hypervillain** [comment]
https://vercel.com/blog/wildcard-domains

**Reply by Utkarshbhimte** [reply]
Were you able to do static generation on each subdomain? Or are you conditionally rendering from the server?

**Reply by tmm** [reply]
I'm curious about this too. I think you are stuck conditionally rendering from the server though.

**Reply by SleepWalker** [reply]
@kevinwgutierrez you should look at `req.headers.host`. It will contain the entire domain, e.g. `foo.example.com`

**Comment by NEDDL** [comment]
@ajaytech, It's been long since this question created but I would appreciate if you could share your solution if found one.
I am interested in the part that redirecting from the user's own custom domain to the wildcard subdomain.
